app/controllers/concerns/verify_sp_attributes_concern.rb

```diff
@@ -46,7 +46,9 @@ def consent_has_expired?
     return false unless sp_session_identity
     return false if sp_session_identity.deleted_at.present?
     last_estimated_consent = sp_session_identity.last_consented_at || sp_session_identity.created_at
-    !last_estimated_consent || last_estimated_consent < Identity::CONSENT_EXPIRATION.ago
+    !last_estimated_consent ||
+      last_estimated_consent < Identity::CONSENT_EXPIRATION.ago ||
+      verified_after_consent?(last_estimated_consent)
```

Comment:
Good catch on expiring consent here.

Reply:
Porta said it might not be needed to show the consent page each time, but this simplified a lot of assumptions to check it like this